saleor/graphql/product/mutations/products.py

```diff
@@ -236,6 +259,21 @@ class Meta:
         description = 'Creates a new product.'
         model = models.Product
 
+    @classmethod
+    def clean_default_variant(cls, cleaned_input, product_type, errors):
+        default_variant = cleaned_input.pop('default_variant', None)
+        if default_variant and product_type.has_variants:
+            cls.add_error(
+                errors, 'default_variant', 'Product type has variants')
+        elif default_variant and not product_type.has_variants:
```

Comment:
That's a bit of a waste to evaluate twice the same conditions. Why not doing
```python
if default_variant:
    if product_type.has_variants:
         pass  # ...
    else:
         pass  # ...
```

It adds another block level, but removes useless CPU instructions (assuming Python does not optimize the code).

Reply:
Totally agree this looks better. Initially I also used nested if blocks, but I had to choose the current implementation because commit didn't pass `codeclimate` check - cognitive complexity of this method was above the threshold (nested if's add additional point to cognitive complexity score).

Reply:
True, but sometimes code written in such manner is much more readable and just feels more natural. There are some places in our code which don't pass this check but in my opinion, this is acceptable if we all agree on readability.